rust/shalom_dart_codegen/dart_tests/test/object_selection/__graphql__/GetListingOpt.shalom.dart

```diff
@@ -0,0 +1,225 @@
+// ignore_for_file: non_constant_identifier_names
+
+class RequestGetListingOpt {
+  /// class memberes
+
+  final GetListingOptListingOpt? listingOpt;
+
+  // keywordargs constructor
+
+  RequestGetListingOpt({this.listingOpt});
+  static RequestGetListingOpt fromJson(Map<String, dynamic> data) {
+    final GetListingOptListingOpt? listingOpt_value;
+    if (data['listingOpt'] != null) {
+      listingOpt_value = GetListingOptListingOpt.fromJson(
+        data['listingOpt'] as Map<String, dynamic>,
+      );
+    } else {
+      listingOpt_value = null;
+    }
+
+    return RequestGetListingOpt(listingOpt: listingOpt_value);
+  }
```

Comment:
```suggestion
  static RequestGetListingOpt fromJson(Map<String, dynamic> data) {
    final GetListingOptListingOpt? listingOpt_value;
    final Map<String, dynamic>? _listingOpt$raw = data['listingOpt'];
    if (_listingOpt$raw!= null) {
        listingOpt_value = GetListingOptListingOpt.fromJson(
          _listingOpt$raw 
        );
    } else {
      listingOpt_value = null;
    }

    return RequestGetListingOpt(listingOpt: listingOpt_value);
  }```
  should generate like this.